tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -11175,7 +11175,9 @@ bool hlsl::ShouldSkipNRVO(clang::Sema &sema, clang::QualType returnType,
     // Check if it's an entry function the hard way
     if (!FD->getDeclContext()->isNamespace() && FD->isGlobal()) {
       // Check if this is an entry function by comparing name
-      if (FD->getName() == sema.getLangOpts().HLSLEntryFunction) {
+      // TODO: Remove this once we put HLSLShaderAttr on all entry functions.
+      if (FD->getIdentifier() &&
```

Comment:
Is this needed for the fix or just an opportunistic clean up?

Reply:
Amusingly it is needed. The new test case asserts on this line without this change because the global operator functions come through here and they don't have valid identifiers.

We didn't hit this in our existing tests because none of our tests tried to declare global operators, and users haven't hit it because we ship our release builds without asserts enabled.